### PR TITLE
Add an option to disable/enable floating IPs

### PR DESCRIPTION
**Pull request by mathias-ioki**
When the OpenStack driver is used, instances get always a public floating IP address. This floating IP is most of the time a public IPv4 address. Public IPv4 addresses are rare and can be very expensive - depending on the OpenStack provider.

This PR allows to disable floating IPs. The default is not changed, so existing deployments don't need an update.

These floating IPs are not required, if IPv6 is available or if a jumphost is already in place, which can be used for the connection to the internal IPs of a VM.

If the option `auto_floating_ip` is not explicit added, it's implicit added with `true` as default value: https://docs.ansible.com/ansible/latest/collections/openstack/cloud/server_module.html#parameter-auto_ip

### CODE REVIEW

**Comment by KevinHonka** [comment]
could we get this merged? I just stumbled over this problem and this would be the solution

**Comment by ednxzu** [comment]
Any update on this ?

**Comment by mathias-ioki** [comment]
To be honest, I don't know how to proceed here. Did I missed something, to get this reviewed?

**Comment by ednxzu** [comment]
I don't know... it looks pretty good to me ? maybe a rebase and a ping to someone able to have a look at it and potentially merge ?

**Comment by apatard** [comment]
@mathias-ioki Sorry. For personal reasons, I've not managed to work on the plugins for a lot of time. This lead to me having to look at a bunch of PR. From what I understand, given that #263 was supposed to supersede this PR and one other, I've merged the other one.
tl;dr it's not your fault and your PR seemed fine. Only, please bear with me.
